Sources/Purchasing/OfferingsManager.swift

```diff
@@ -80,9 +81,24 @@ class OfferingsManager {
             switch result {
             case let .success(response):
                 self.handleOfferingsBackendResult(with: response,
+                                                  appUserID: appUserID,
                                                   fetchPolicy: fetchPolicy,
                                                   completion: completion)
 
+            case let .failure(.networkError(networkError)) where networkError.isServerDown:
+                Logger.warn(Strings.offering.fetching_offerings_failed_server_down)
+
+                // If unable to fetch offerings when server is down, attempt to load them from disk cache.
+                self.fetchCachedOfferingsFromDisk(appUserID: appUserID,
+                                                  fetchPolicy: fetchPolicy) { offerings in
+                    if let offerings = offerings {
+                        self.dispatchCompletionOnMainThreadIfPossible(completion, value: .success(offerings))
+                    } else {
+                        self.handleOfferingsUpdateError(.backendError(.networkError(networkError)),
```

Comment:
If it fails or we didn't have anything in the cache, we forward the original error.